src/scipp/core/reduction.py

```diff
@@ -107,6 +107,42 @@ def nanmean(x: VariableLikeType, dim: str | None = None) -> VariableLikeType:
         return _call_cpp_func(_cpp.nanmean, x, dim=dim)  # type: ignore[return-value]
 
 
+def average(x: VariableLikeType, dim: str | None = None, weights: VariableLikeType | None = None) -> VariableLikeType:
+    """(Weighted) average of the elements in the input. 
+
+    Where, `weights = None` this is equivalent to :py:func`scipp.mean`. 
+
+    Parameters
+    ----------
+    x: scipp.typing.VariableLike
+        Input data.
+    dim:
+        Dimension along which to calculate the mean. If not
+        given, the nanmean over all dimensions is calculated.
+
+    weights: scipp.typing.VariableLike
+        Array of weights for each value in `x`.
+
+    Returns
+    -------
+    : Same type as x
+        The (weighted) average of the input values.
+
+    See Also
+    --------
+    scipp.nanvar:
+        Compute the variance, ignoring NaN's.
+    scipp.nanstd:
+        Compute the standard deviation, ignoring NaN's.
+    scipp.mean:
+        Compute the mean without special handling of NaN.
+    """
+    if weights is None:
+        return mean(x=x, dim=dim)
+    else:
+        return (weights * x).sum() / weights.sum()
```

Comment:
As the weights show up twice, I am not sure this will automatically yield the correct variances. Should be easy to check by doing the derivatives by hand?

Reply:
I think, I need to manually compute the following (https://en.wikipedia.org/wiki/Weighted_arithmetic_mean#Weighted_sample_variance). 

So that the result matches with the `sc.mean`, do you want me to then divide by $\sqrt{N}$?

Reply:
I was yeah.